mlflow/utils/annotations.py

```diff
@@ -21,15 +22,23 @@ def deprecated(alternative=None, since=None):
     :param func: A function to mark
     :returns Decorated function.
     """
-    def deprecated_func(func):
-        since_str = " since %s" % since if since else ""
-        notice = ".. Warning:: Deprecated%s: This method will be removed in " % since_str + \
-                 "a near future release."
-        if alternative is not None and alternative.strip():
-            notice += " Use ``%s`` instead." % alternative
-        func.__doc__ = notice + "\n" + func.__doc__
-        return func
-    return deprecated_func
+
+    def deprecated_decorator(func):
+        @wraps(func)
+        def deprecated_func(*args, **kwargs):
+            since_str = " since %s" % since if since else ""
+            notice = ".. Warning::" + \
```

Comment:
Nit: Can we define notice like this
```
            notice = (
                ".. Warning:: {function_name} is deprecated{since_string}. This method will be" 
                " removed in a near future release".format(
                    function_name=func.__name__,
                    since_string=since_str)
            )
```?

The current construction with string concatenation that uses formatting in the middle is pretty hard to read.